includes/reader-revenue/my-account/class-woocommerce-my-account.php

```diff
@@ -514,6 +522,128 @@ public static function show_message_after_logout() {
 			WooCommerce_Connection::add_wc_notice( __( 'You have successfully logged out.', 'newspack-plugin' ), 'success' );
 		}
 	}
+
+	/**
+	 * Adds option to display memberships without subscriptions on the Subscriptions tab of My Account..
+	 *
+	 * @param array $settings WooCommerce Memberships settings.
+	 * @return array
+	 */
+	public static function option_display_memberships_without_subs( $settings ) {
+		$settings[] = array(
+			'name' => __( 'Newspack Reader Activation', 'newspack-plugin' ),
+			'type' => 'title',
+			'id'   => 'wc_memberships_show_unsubbed_memberships',
+		);
+
+		$settings[] = array(
+			'type'    => 'checkbox',
+			'id'      => 'wc_memberships_show_unsubbed_memberships',
+			'name'    => __( 'Memberships without subscriptions', 'newspack-plugin' ),
+			'desc'    => __( 'Display memberships that don\'t have active subscriptions on the My Account Subscriptions tab.', 'newspack-plugin' ),
+			'default' => 'no',
+		);
+
+		$settings[] = array(
+			'type' => 'sectionend',
+		);
+
+		return $settings;
+	}
+
+	/**
+	 * Check if a reader has memberships that aren't associated with subscriptions.
+	 *
+	 * @return array
+	 */
+	public static function get_memberships_without_subs() {
+		if ( function_exists( 'wc_memberships_get_user_active_memberships' ) ) {
+			$customer_id              = \get_current_user_id();
+			$memberships_info         = wc_memberships_get_user_active_memberships( $customer_id );
```

Comment:
Nit: this is not currently throwing any errors for me, but it's something we've run into occasionally, so this is a good practice to get into the habit of:

```suggestion
			$memberships_info         = \wc_memberships_get_user_active_memberships( $customer_id );
```

If you're working in a PHP file that has a namespace (as [this one does](https://github.com/Automattic/newspack-plugin/blob/trunk/includes/reader-revenue/my-account/class-woocommerce-my-account.php#L8)), when calling any global function, it's a good idea to prefix it with `\`. This tells PHP to look for this function in the global namespace. If you don't include the `\` prefix, then PHP will first look for the function in the current namespace and then fall back to global if it doesn't exist in the current namespace. I've seen instances where this can cause fatal errors, so adding the `\` sidesteps that whole issue and also makes the code a little clearer in a namespaced file (even if it's not required to avoid errors like in this case).

The exception to this rule is [internal PHP functions](https://www.php.net/manual/en/functions.internal.php), so if you can find the function in the PHP manual, you don't need to specify a namespace (global or otherwise) for it. In my code editor internal PHP functions are helpfully highlighted in a different color than other functions:

<img width="428" alt="Screenshot 2024-04-16 at 1 03 47 PM" src="https://github.com/Automattic/newspack-plugin/assets/2230142/291a3576-8284-494e-9227-0228c9807ebc">

You also don't need to prefix the function name when checking whether it exists (like `function_exists`) as this will check for the function in every namespace including the global one, unless you pass a specific namespace.

Reply:
TIL -- nice! I didn't realize the purpose of the `\`, I figured it had something to do with namespaced/not namespaced, but didn't realize what it actually did under the hood.